[Ooi Wen Ree] iP

Wrooi opened the pull request.

Wrooi committed:
Rename chatbot to Rudeus and update personality

Wrooi committed:
Update Rudeus to echo user input & exit after 'bye' command

Wrooi committed:
Update Rudeus's responses to include indents & print borders

Wrooi committed:
Refactor: Add readAndProcessUserInput & printGreetingMessage

Wrooi committed:
Update to logo

Wrooi committed:
Add task tracking & listing based on user input

Wrooi committed:
Add Task Class & ability to mark & unmark list

Wrooi committed:
Remove redundant arguement from printMessageWithBorders

sanjay-shiva-kumar commented on src/main/java/Rudeus.java:
```diff
@@ -0,0 +1,114 @@
+    private static final Task[] taskList = new Task[100]; // Array to store tasks
+    private static int taskCount = 0; // Counter for tasks
+
+    // Method to automatically print with indents
```
Shouldn't the comments for public functions be done in javadoc format?

sanjay-shiva-kumar commented on src/main/java/Rudeus.java:
```diff
@@ -0,0 +1,114 @@
+    }
+
+    // Method to read and process user input
+    public static void readAndProcessUserInput() {
```
I like how you made a separate dedicated function for reading and processing user inputs 👍

sanjay-shiva-kumar commented on src/main/java/Rudeus.java:
```diff
@@ -0,0 +1,114 @@
+    private static int taskCount = 0; // Counter for tasks
+
+    // Method to automatically print with indents
+    private static void printWithIndents(String message) {
+        String indent = " ".repeat(MAX_INDENT_LEVEL); // 4 spaces per indent level
+        System.out.println(indent + message);
+    }
+
+    // Method to automatically print messages with borders
+    private static void printMessageWithBorders(String message) {
+        String border = "──────────────────────────────────────────────────────────────────────────";
+            printWithIndents(border);
+            printWithIndents(message);
+            printWithIndents(border);
+    }
```
I like how you added formatting functions for printing 👍

sanjay-shiva-kumar commented on src/main/java/Rudeus.java:
```diff
@@ -0,0 +1,114 @@
+        printMessageWithBorders(taskListMessage.toString().trim());
+    }
+
+    private static void markTask(int index, boolean isMark) {
+        String extraIndent = " ".repeat(MAX_INDENT_LEVEL + 2); // 6 spaces per indent level
+        if (index < 0 || index >= taskCount) {
+            printMessageWithBorders("Invalid task number.");
+            return;
+        }
+        if (isMark) {
+            taskList[index].setIsDone(true);
+            printMessageWithBorders("Nice! I've marked this task as done:\n" + extraIndent + taskList[index]);
+        } else {
+            taskList[index].setIsDone(false);
+            printMessageWithBorders("OK, I've marked this task as not done yet:\n" + extraIndent + taskList[index]);
+        }
+    }
```
I like how you made a function for marking tasks 👍

BenyAlbatross replied:
Could the name of this `markTask` method be confusing, since it accepts a boolean parameter that controls either marking or unmarking? Also, perhaps you could add exception handling if the user tries to mark a task that has already been marked?

Yxiang-828 commented on src/main/java/Rudeus.java:
```diff
@@ -80,20 +74,20 @@ public static void readAndProcessUserInput() {
                     markTask(idx, false);
                 } catch (NumberFormatException e) { // Handle invalid number format
-                    printMessageWithBorders("Please provide a valid task number to unmark.", true);
+                    printMessageWithBorders("Please provide a valid task number to unmark.");
```
nice catch to handle exceptions

Yxiang-828 commented on src/main/java/Rudeus.java:
```diff
@@ -2,6 +2,8 @@
 
 public class Rudeus {
     private static final int MAX_INDENT_LEVEL = 4; // Maximum indent level
+    private static final Task[] taskList = new Task[100]; // Array to store tasks
```
i recommend not using fixed size arrays. A dynamic one or vector would be more effective for potentially large size of tasks

Yxiang-828 reviewed (commented):
well done in overall. I suggest start making class files for tasks, mark/unmark etc. because there will be way more features coming in soon, cluttering in one file will not be an optimal option

BenyAlbatross commented on src/main/java/Rudeus.java:
```diff
@@ -0,0 +1,114 @@
+                        "│░█████████  ░██     ░██ ░██    ░██ ░█████████  ░██     ░██  ░████████ │\n" +
+                        "│░██   ░██   ░██     ░██ ░██    ░██ ░██         ░██     ░██         ░██│\n" +
+                        "│░██    ░██   ░██   ░██  ░██   ░██  ░██          ░██   ░██   ░██   ░██ │\n" +
+                        "│░██     ░██   ░██████   ░███████   ░██████████   ░██████     ░██████  │\n" +
+                        "└──────────────────────────────────────────────────────────────────────┘\n";
+        System.out.println("Yo! The name's \n" + logo + "At your service, as always. Need some magic—or maybe just a hand? Ask away!\n" +
+                "──────────────────────────────────────────────────────────────────────────");
+    }
+
+    public static void main(String[] args) {
+        printGreetingMessage();
+        readAndProcessUserInput();
+    }
+}
```
Perhaps the main method should include exception handling to prevent the application from crashing unexpectedly due to input errors or other runtime exceptions?

BenyAlbatross commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,18 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Should the `description` and `isDone` fields be set to private, and for public getters and setters to be provided instead?

BenyAlbatross commented on src/main/java/Rudeus.java:
```diff
@@ -0,0 +1,114 @@
+import java.util.Scanner;
+
+public class Rudeus {
+    private static final int MAX_INDENT_LEVEL = 4; // Maximum indent level
```
Defining MAX_INDENT_LEVEL as a constant increases the maintainability of your code, good job.

BenyAlbatross reviewed (commented):
Overall looks good, your code is readable and I like how you separated out your tasks into methods.

Wrooi committed:
Docs: Add documentations in javadoc format

Wrooi committed:
Refactor: Rename markTask method

Wrooi committed:
Refactor: Changed fixed array of 100 to vector

Wrooi committed:
Refactor: Change the fixed array of 100 to a vector

Wrooi committed:
Chore: Add new Deadilne, Event and Todo Classes

Wrooi committed:
Merge branch 'master' of https://github.com/Wrooi/ip

Wrooi committed:
Docs: Add CheckStyle and documentations in Javadoc format

Wrooi committed:
Feat: Update Deadline, Event, Todo Classes

Wrooi committed:
Feat: Update Rudeus.java addTask method

Wrooi committed:
Test: Add test to Text UI Testing

Wrooi committed:
Refactor: Separate methods from Rudeus to Parser, TaskManager and Ui Classes

Wrooi committed:
Refactor: Separate methods in Rudeus to CommandManager Class and use Enums

Wrooi committed:
Docs: Add documentation to methods in various files

Wrooi committed:
Bug: Add Exception Handling Consistency to CommandManager Class

Wrooi committed:
Docs: Add CheckStyle documentation to various files

Wrooi committed:
Chore: Add getter method for description in Task Class

Wrooi committed:
Perf: Add explicit empty argument checks to avoid unnecessary exceptions

Wrooi committed:
Feat: Add personalised error handling for user inputs in CommandManager Class

Wrooi committed:
Feat: Add personalised error handling for user inputs in Parser Class

Wrooi committed:
Feat: Add personalised error handling for user inputs in TaskManager Class

Wrooi committed:
Refactor: Move Classes into Packages

Wrooi committed:
Bug: Update Text UI Testing Path

Wrooi committed:
Feat: Make Task Class Abstract

Wrooi committed:
Test: Update input test cases

Wrooi committed:
Merge branch 'branch-Level-5'

Wrooi committed:
Feat: Add personalised Delete Task Method

Wrooi committed:
Fix: Fix indentation bug for Delete Method

Wrooi committed:
Test: Update input test cases to test delete

Wrooi committed:
Feat: Change Vector to ArrayList for storing tasks

Wrooi committed:
Feat: Add storage class for saving tasks

Wrooi committed:
Feat: Add Methods in Storage Class to load the Data from tasks.txt

Wrooi committed:
Fix: Fix a bug in the directory for Text UI testing for save/load

Wrooi committed:
Refactor: Separate Storage Class into LocalSave and TaskSerializer

Wrooi committed:
Merge branch 'branch-Level-6'

Wrooi committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/rudeus/task/TaskManager.java

Wrooi committed:
Feat: Add jar Manifest File

Wrooi committed:
Refactor: Make TaskManager an object-oriented class with instance fields and methods
# Update CommandManager to use TaskManager as an instance, removed record usage.
# Update Rudeus.java to instantiate TaskManager and CommandManager objects.
# Ensure all task operations and command handling use object instances for better encapsulation and extensibility.

Wrooi committed:
Merge branch 'branch-A-MoreOOP'

Wrooi committed:
Feat: Add Find command

Wrooi committed:
Test: Update input test cases to test find command

Wrooi committed:
Merge branch 'branch-Level-9'

Wrooi committed:
Docs: Add JavaDoc comments to all files

Wrooi committed:
Merge branch 'branch-A-JavaDoc'

Wrooi committed:
Docs: Update README.md with User Guides

Wrooi committed:
Merge branch 'branch-A-UserGuide'

Wrooi committed:
Merge pull request #1 from Wrooi/branch-A-UserGuide

Docs: Update README.md with User Guides

Wrooi committed:
Merge commit '41cbc062f12f2201b2f868df9d574bf59a470c19'

Wrooi committed:
Docs: Update User Guide

Wrooi committed:
Merge branch 'branch-A-UserGuide'

Wrooi committed:
Docs: Update User Guide FINAL

Wrooi committed:
Merge branch 'branch-A-UserGuide'

Wrooi committed:
Feat: Update jar Manifest File
